qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,698 @@
+        binding_dict = {}
+        for ch in schedule.channels:
+            if ch.is_parameterized():
+                binding_dict[ch.index] = self._get_channel_index(qubits, ch)
+
+        # Binding the channel indices makes it easier to deal with parameters later on
+        schedule = schedule.assign_parameters(binding_dict, inplace=False)
+
+        # The following code allows us to get the keys when the schedule has call instructions.
+        # We cannot inline the subroutines yet because we would lose the name of the subroutines.
+        parameter_keys = Calibrations.get_parameter_keys(schedule, set(), binding_dict, qubits)
+
+        # Now that we have the parameter keys we must inline all call subroutines.
+        schedule = inline_subroutines(schedule)
+
+        # Build the parameter binding dictionary.
+        free_params = free_params if free_params else []
+
+        for key in parameter_keys:
+            if key.parameter not in free_params:
+                # Get the parameter object. Since we are dealing with a schedule the name of
+                # the schedule is always defined. However, the parameter may be a default
+                # parameter for all qubits, i.e. qubits may be None.
+                if key in self._parameter_map:
+                    param = self._parameter_map[key]
+                elif (key.schedule, key.parameter, None) in self._parameter_map:
+                    param = self._parameter_map[(key.schedule, key.parameter, None)]
+                else:
+                    raise CalibrationError(
+                        f"Ill configured calibrations {key} is not present and has not default value."
+                    )
+
+                if param not in binding_dict:
+                    binding_dict[param] = self.get_parameter_value(
+                        key.parameter, key.qubits, key.schedule, group=group
+                    )
+
+        return schedule.assign_parameters(binding_dict, inplace=False)
+
+    @staticmethod
+    def get_parameter_keys(
+        schedule: Schedule, keys: Set, binding_dict: Dict[Parameter, int], qubits: Tuple[int, ...]
+    ):
+        """
+        Recursive function to extract parameter keys from a schedule. The recursive
+        behaviour is needed to handle Call instructions. Each time a Call is found
+        get_parameter_keys is call on the subroutine of the Call instruction and the
+        qubits that are in the subroutine. This also implies carefully extracting the
+        qubits from the subroutine and in the appropriate order.
+
+        Args:
+            schedule: A schedule from which to extract parameters.
+            keys: A set of keys that will be populated.
+            binding_dict: A binding dictionary intended only for channels. This is needed
+                because calling assign_parameters on a schedule with a Call instruction will
+                not assign the parameters in the subroutine of the Call instruction.
```

Comment:
This is not true. Parameters in the `Call` instructions are stored in `.arguments` table, i.e. this instruction separately stores the program and arguments. The stored parameters are automatically assigned when your inline schedules. This means if you check `inst.subroutine.parameters` no parameter seems to be assigned, but actually this is stored in the `inst.arguments`. If this doesn't work let me know with code example, should be some bug.

Reply:
There might be a bug in the way parameters are assigned when there are call instructions. E.g.
```python
from qiskit.circuit import Parameter
import qiskit.pulse as pulse
from qiskit.pulse import DriveChannel, Gaussian
d0_ = Parameter("ch0")

with pulse.build(name="xp") as xp:
    pulse.play(Gaussian(160, 0.1, 40), DriveChannel(d0_))

with pulse.build(name="xp2") as xp2:
    pulse.call(xp)

print(xp2)
print(xp2.assign_parameters({d0_: 2}))
```

Reply:
This is fixed in 3006695 by using `assigned_subroutine`.